Modules/Core/SpatialObjects/include/itkSpatialObjectProperty.h

```diff
@@ -111,7 +105,7 @@ class ITKSpatialObjects_EXPORT SpatialObjectProperty
   void
   SetTagScalarValue(const std::string & tag, double value);
   void
-  SetTagStringValue(const std::string & tag, const std::string & value);
+  SetTagStringValue(const std::string & tag, std::string value);
```

Comment:
Do I understand correctly that the SWIG problem occurred with the parameter `value`, but not with `tag`? If so, what would be the general guideline on when to pass `std::string` by-value, and when to pass it by `const &`?

Reply:
I have no idea. :)   We have only 1 data point at this time.   SWIG seems to be somehow converting this function to a new version that returns a std::string * (rather than returning a void as defined by the C++ code).   In another comment,, I reference documentation (albeit from an old version of SWIG, but perhaps still relevant) that suggests that SWIG supports wrapping templates that could be doing this conversion for Set functions that take multiple args.   Furthermore, other documentation explains why returning std::string & might be converted to std::string * in the Get cases.  I suspect there is a bug in our SWIG wrapping specification for these reasons and because new SWIG should handled std::string fine and these wonkynesses shouldn't be happening.  I'll file an issue.